### PR TITLE
U541/t682 soft shadows beginning

### PR DESCRIPTION
**Pull request by Piratkopia13**
Soft shadows! sloow :c hard shadows faaast :)
Also changed how metaballs are rendered

Note: branch is named U541/T554-ShadowShading

### CODE REVIEW

**Comment by h3nx** [comment]
i pray

File: Sail/src/API/DX12/renderer/DX12RaytracingRenderer.cpp [hunk 1]
```diff
@@ -163,51 +196,227 @@ void DX12RaytracingRenderer::present(PostProcessPipeline* postProcessPipeline, R
...
+	m_shadedOutput->transitionStateTo(cmdList, D3D12_RESOURCE_STATE_RENDER_TARGET);
+	m_outputBloomTexture->transitionStateTo(cmdList, D3D12_RESOURCE_STATE_RENDER_TARGET);
+
+	// Bind shaded and bloom outputs
+	D3D12_CPU_DESCRIPTOR_HANDLE rtvHandles[2] {
+		m_shadedOutput->getRtvCDH(),
+		m_outputBloomTexture->getRtvCDH()
+	};
+	cmdList->OMSetRenderTargets(2, rtvHandles, false, nullptr);
+	//cmdList->OMSetRenderTargets(1, &m_shadedOutput->getRtvCDH(), false, nullptr);
```
Comment (whalemane) [inline Sail/src/API/DX12/renderer/DX12RaytracingRenderer.cpp]: Remove this?

File: Sail/src/API/DX12/resources/DX12RenderableTexture.cpp [hunk 2]
```diff
@@ -31,7 +34,7 @@ DX12RenderableTexture::DX12RenderableTexture(UINT aaSamples, unsigned int width,
 		m_dsvHeapCDHs[i] = m_cpuDsvDescHeap.getCPUDescriptorHandleForIndex(i);
 	}
 	createTextures();
-	renameBuffer(name);
+	//renameBuffer(name);
```
Comment (whalemane) [inline Sail/src/API/DX12/resources/DX12RenderableTexture.cpp]: Remove this?

File: Sail/src/Sail/entities/EntityFactory.cpp [hunk 3]
```diff
@@ -64,7 +64,7 @@ void EntityFactory::CreateCandle(Entity::SPtr& candle, const glm::vec3& lightPos
 	PointLight pl;
 	pl.setColor(glm::vec3(0.55f, 0.5f, 0.45f));
 	pl.setPosition(glm::vec3(lightPos.x, lightPos.y + .5f, lightPos.z));
-	pl.setAttenuation(0.f, 0.f, 0.2f);
+	pl.setAttenuation(10.f, 0.f, 0.2f);
```
Comment (whalemane) [inline Sail/src/Sail/entities/EntityFactory.cpp]: Is this correct or is this change due to some testing?

File: Sail/src/Sail/entities/systems/Gameplay/LevelSystem/LevelSystem.cpp [hunk 4]
```diff
@@ -1760,7 +1760,7 @@ void LevelSystem::addClutterModel(const std::vector<Model*>& clutterModels, Mode
 		AudioComponent* ac = e2->addComponent<AudioComponent>();
 		sc->light.setColor(glm::vec3(1.0f, 0.2f, 0.0f));
 		sc->light.setPosition(glm::vec3(0, tileHeight * 5 - 0.05, 0));
-		sc->light.setAttenuation(1.f, 0.01f, 0.01f);
+		sc->light.setAttenuation(30.f, 0.01f, 0.01f);
```
Comment (whalemane) [inline Sail/src/Sail/entities/systems/Gameplay/LevelSystem/LevelSystem.cpp]: Is this correct or change for some testing?

File: Sail/src/Sail/utils/Storage/SettingStorage.cpp [hunk 5]
```diff
@@ -258,7 +258,7 @@ void SettingStorage::createGameDefaultMap() {
 	gameSettingD["tileSize"] =	DynamicSetting(7.0f, 1.0f, 30.0f);
 	gameSettingD["clutter"] = DynamicSetting(0.85f,	0.0f,	1.0f);
 	gameSettingD["seed"] =    DynamicSetting(0.0f,	0.0f,	1000000.0f);
-	gameSettingD["sprinklerTime"] = DynamicSetting(75.0f, 0.0f, 600.0f);
+	gameSettingD["sprinklerTime"] = DynamicSetting(10.0f, 0.0f, 600.0f);
```
Comment (whalemane) [inline Sail/src/Sail/utils/Storage/SettingStorage.cpp]: This should probably be changed back

File: Sail/src/Sail/entities/EntityFactory.cpp (hunk 3 above)
Comment (Piratkopia13) [reply Sail/src/Sail/entities/EntityFactory.cpp]: Yes it is correct. Altough the first attenuation value is not used as attenuation anymore, it specifies the light radius. I will make some changes to make this more clear!

**Comment by h3nx** [comment]
Gameplay seems fine.
Shadows be shadows, switching is seamless.
killcam broken

**Comment by Piratkopia13** [comment]
Viktors commits should have fixed the killcam bug! Please test and change labels when possible

**Pull request merged**